test/OpenSpaceToolkit/Astrodynamics/Flight/System/PropulsionSystem.test.cpp

```diff
@@ -0,0 +1,223 @@
+
+        const PropulsionSystem propulsionSystem = {thrust, specificImpulse};
+
+        EXPECT_TRUE(propulsionSystem == propulsionSystem_);
+    }
+
+    {
+        const Scalar thrust = Scalar(0.02, thrustSIUnit);
+        const Scalar specificImpulse = Scalar(100.0, specificImpulseSIUnit);
+
+        const PropulsionSystem propulsionSystem = {thrust, specificImpulse};
+
+        EXPECT_FALSE(propulsionSystem == propulsionSystem_);
+    }
+
+    {
+        const Scalar thrust = Scalar(0.01, thrustSIUnit);
+        const Scalar specificImpulse = Scalar(100.45, specificImpulseSIUnit);
+
+        const PropulsionSystem propulsionSystem = {thrust, specificImpulse};
+
+        EXPECT_FALSE(propulsionSystem == propulsionSystem_);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem, NotEqualToOperator)
+{
+    {
+        const Scalar thrust = Scalar(0.01, thrustSIUnit);
+        const Scalar specificImpulse = Scalar(100.0, specificImpulseSIUnit);
+
+        const PropulsionSystem propulsionSystem = {thrust, specificImpulse};
+
+        EXPECT_FALSE(propulsionSystem != propulsionSystem_);
+    }
+
+    {
+        const Scalar thrust = Scalar(0.02, thrustSIUnit);
+        const Scalar specificImpulse = Scalar(100.0, specificImpulseSIUnit);
+
+        const PropulsionSystem propulsionSystem = {thrust, specificImpulse};
+
+        EXPECT_TRUE(propulsionSystem != propulsionSystem_);
+    }
+
+    {
+        const Scalar thrust = Scalar(0.01, thrustSIUnit);
+        const Scalar specificImpulse = Scalar(100.45, specificImpulseSIUnit);
+
+        const PropulsionSystem propulsionSystem = {thrust, specificImpulse};
+
+        EXPECT_TRUE(propulsionSystem != propulsionSystem_);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem, IsDefined)
+{
+    EXPECT_TRUE(propulsionSystem_.isDefined());
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem, StreamOperator)
+{
+    {
+        testing::internal::CaptureStdout();
+
+        EXPECT_NO_THROW(std::cout << propulsionSystem_ << std::endl);
+
+        EXPECT_FALSE(testing::internal::GetCapturedStdout().empty());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem, Print)
+{
+    {
+        testing::internal::CaptureStdout();
+
+        EXPECT_NO_THROW(propulsionSystem_.print(std::cout, true));
+        EXPECT_NO_THROW(propulsionSystem_.print(std::cout, false));
+        EXPECT_FALSE(testing::internal::GetCapturedStdout().empty());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem, GetThrust)
+{
+    {
+        EXPECT_TRUE(propulsionSystem_.getThrust() == thrust_);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem, GetSpecificImpulse)
+{
+    {
+        EXPECT_TRUE(propulsionSystem_.getSpecificImpulse() == specificImpulse_);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem, GetMassFlowRate)
```

Comment:
Maybe a test case where the `PropulsionSystem` is not created with the go-to thrust and specific impulse units, but some other ones so that we test the construction transformation to compute the `massFlowRate` (this only makes sense if my comment on this above makes sense)